app/views/play/level/PlayLevelView.coffee

```diff
@@ -835,19 +835,23 @@ module.exports = class PlayLevelView extends RootView
       when @level?.isType('web-dev') then windowHeight - controlBarHeight
       when tomeLocation is 'bottom' then Math.min(windowHeight - minTomeHeight - controlBarHeight, windowWidth / canvasAspectRatio)
       else Math.min(windowHeight - (if controlBarLocation is 'left' then controlBarHeight else 0), (windowWidth - minCodeWidth - minWorkspaceWidth - minToolboxWidth) / canvasAspectRatio)
+    desiredCanvasWidth = canvasHeight * canvasAspectRatio
     if me.get('aceConfig').preferWideEditor or features?.china
-      desiredCanvasWidth = canvasHeight * canvasAspectRatio
       if windowWidth - desiredCanvasWidth < 500 and tomeLocation is 'right'
         # windowWidth / 1.82 get 55% of the screen width for canvas -- our old style
         # windowWidth - 500 get 500px for editor so won't get a really narrow editor
         canvasWidth =  Math.max(windowWidth / 1.82, windowWidth - 500)
       else
         canvasWidth = desiredCanvasWidth
+      if @$el.hasClass('real-time')
```

Comment:
Check that the `@level?.isType('game-dev')` also, there are other cases when we go real-time.